{
  Exception caughtException=null;
  try {
    din.close();
  }
 catch (  Exception err) {
    LOG.error("Error closing input stream:" + err.getMessage(),err);
    caughtException=err;
  }
  if (client != null) {
    try {
      client.close();
    }
 catch (    Exception err) {
      LOG.error("Error closing client:" + err.getMessage(),err);
      caughtException=(caughtException == null ? err : caughtException);
    }
  }
  if (caughtException != null) {
    throw new IOException("Exception during close: " + caughtException.getMessage(),caughtException);
  }
}
